packages/core/src/resolve.ts

```diff
@@ -115,6 +115,9 @@ export class BaseResolver {
         const { body, mimeType } = await readFileFromUrl(absoluteRef, this.config.http);
         return new Source(absoluteRef, body, mimeType);
       } else {
+        if (fs.lstatSync(absoluteRef).isDirectory()) {
+          throw new Error(`API path cannot refer to a folder. ${absoluteRef} is a folder`);
```

Comment:
This is not only about API. This error can happen for any $ref (because you put it into generic resovle).

Either move this code somewhere else or update the message.